runtime/doc/lua.txt

```diff
@@ -1738,6 +1738,33 @@ pesc({s})                                                         *vim.pesc()*
     See also: ~
       • https://github.com/rxi/lume
 
+ringbuf({size})                                                *vim.ringbuf()*
+    Create a ring buffer limited to a maximal number of items. Once the buffer
+    is full, adding a new entry overrides the oldest entry.
+>
+
+    local ringbuf = vim.ringbuf(4)
+    ringbuf.add("a")
+    ringbuf.add("b")
+    ringbuf.add("c")
+    ringbuf.add("d")
+    ringbuf.add("e")      -- overrides "a"
+    print(ringbuf.poll()) -- returns "b"
+    print(ringbuf.poll()) -- returns "c"
+<
+
+    Returns a table with the following functions:
+    • add(item: T): Adds an item to the buffer
+    • clear(): Removes all items from the buffer
+    • peek(): Returns the last unread item, without removing it
+    • poll(): Removes and returns the last unread item
```

Comment:
Would `get()` or `pop()` be better?

Reply:
I'd go with `pop()` and `push()` then. `put` is used in some languages to set a given key. `map.put("x", 1)` - where you expect the same call to override the item with the same key. The operation here behaves a bit differently.

`get()` imho doesn't signal the "removes the item" behavior.

Reply:
How about `read()` and `write()`?

Reply:
Oh, it only adds/removes one item at a time. The I guess `push()` and `pop()` are better.